src/main/java/Duke.java

```diff
@@ -1,10 +1,196 @@
+        String[] splitByForwardSlash = rawInput.split("/",2);
+        decoded[TASK_DATA_INDEX_DESCRIPTION] = splitByForwardSlash[0];
+        String[] splitBySpace = splitByForwardSlash[1].split(" ",2);
+        decoded[TASK_DATA_INDEX_ADDITIONAL_INFO] = splitBySpace[1];
+        return decoded;
+    }
+
+
+    private static String getAdditionalInfo(String[] decodedInput) {
+        return decodedInput[1];
+    }
+
+    private static String getDescription(String[] decodedInput) {
+        return decodedInput[0];
+    }
+
+    private static void executeDeadlineTask(String rawInput) {
+        String[] decodedInput = decodeInput(rawInput);
+        String description = getDescription(decodedInput);
+        String by = getAdditionalInfo(decodedInput);
+        createDeadlineTask(description, by);
+    }
+
+    private static void createDeadlineTask(String description, String by) {
+        taskList[listCount] = new Deadline(description, by);
+        showSuccessfulAdd();
+        listCount++;
+    }
+
+    private static void executeTodoTask(String todoInput) {
+        createTodoTask(todoInput);
+    }
+
+    private static void createTodoTask(String todoInput) {
+        taskList[listCount] = new Todo(todoInput);
+        showSuccessfulAdd();
+        listCount++;
+    }
+
+    private static void executeCompleteTask(String taskIndexString) {
+        int taskIndex = Integer.parseInt(taskIndexString) - 1;
+        Task currentTask = taskList[taskIndex];
+        currentTask.setDone();
+        showSuccessfulComplete(currentTask);
+    }
+
+    private static void showSuccessfulComplete(Task currentTask) {
+        System.out.println("Nice! You did the following task:"
+                + "\n [" + currentTask.getStatusIcon() + "] "
+                + currentTask.getDescription());
+    }
+
+    private static void executeListTasks() {
+        showAllTasks();
+    }
+
+    private static void showAllTasks() {
+        System.out.println("Here are the tasks in your list:");
+        int itemCount = 1;
+        for (Task item : taskList) {
+            if (item != null) {
+                System.out.println(itemCount + "." + item);
+                itemCount++;
+            }
+        }
+    }
+
+    private static String[] splitUserCommand(String userCommand) {
+        final String[] split = userCommand.trim().split(" ",2);
+        if (split.length >= 2) {
+            return split;
+        } else {
+            return new String[] {split[0],""};
+        }
+    }
+
+    private static String getInput() {
+        System.out.print("What would you like me to do? ");
+        Scanner in = new Scanner(System.in);
+        String userInput = in.nextLine();
+        while (userInput.trim().isEmpty()) {
+            userInput = in.nextLine();
+        }
+        return userInput;
+    }
+
+    private static void initTaskList() {
+        taskList = new Task[100];
+        listCount = 0;
+    }
```

Comment:
You could think about using a TaskManager class to manage this task array that contains all your tasks. You can refer to the lender code that was discussed in class for a similar concept.